src/tactic/positivity.lean

```diff
@@ -363,4 +367,39 @@ meta def positivity_abs : expr → tactic strictness
   nonnegative <$> mk_app ``abs_nonneg [a] -- else report nonnegativity
 | _ := failed
 
+private alias nat.cast_pos ↔ _ nat_cast_pos
+
+private lemma int_coe_nat_nonneg (n : ℕ) : 0 ≤ (n : ℤ) := n.cast_nonneg
+private lemma int_coe_nat_pos {n : ℕ} : 0 < n → 0 < (n : ℤ) := nat.cast_pos.2
+
+private lemma int_cast_nonneg [ordered_ring α] {n : ℤ} (hn : 0 ≤ n) : 0 ≤ (n : α) :=
+by { rw ←int.cast_zero, exact int.cast_mono hn }
+
+private alias int.cast_pos ↔ _ int_cast_pos
+
+private alias rat.cast_nonneg ↔ _ rat_cast_nonneg
+private alias rat.cast_pos ↔ _ rat_cast_pos
+
+/-- Extension for the `positivity` tactic: casts from `ℕ`, `ℤ`, `ℚ`. -/
+@[positivity]
+meta def positivity_coe : expr → tactic strictness
+| `(@coe _ %%typ %%inst %%a) := do
+  match inst with
```

Comment:
Idem.

Reply:
How am I supposed to use `guard` here given that there are several branches?

Reply:
Something like:
```
(do unify inst `(@coe_to_lift _ _ nat.cast_coe)
    ...) <|>
(do unify inst `(@coe_to_lift _ _ int.cast_coe)
    ...) <|>
(do unify inst `(@coe_to_lift _ _ rat.cast_coe)
    ...) <|>
(do unify inst `(@coe_to_lift _ _ $ @coe_base _ _ int.has_coe)
    ...)
```

Reply:
That doesn't work. I get a `don't know how to synthesize placeholder` on the first underscore in `@coe_to_lift _ _`.

Reply:
Alright, since we can't figure out a nicer way to do this, and it might turn out that we don't want it anyway, let's just leave a comment and explain how to fix a resulting failure if it happens to pop up later.
```suggestion
  -- TODO: Using `match` here might turn out too strict since we really want the instance to *unify*
  -- with one of the instances below rather than being equal on the nose.
  -- If this turns out to indeed be a problem, we should figure out the right way to pattern match
  -- up to defeq rather than equality of expressions.
  -- See also "Reflexive tactics for algebra, revisited" by Kazuhiko Sakaguchi at ITP 2022.
  match inst with
```